cdist/exec/local.py

```diff
@@ -238,7 +239,7 @@ def run_script(self, script, env=None, return_output=False,
                         message_prefix=message_prefix)
 
     def save_cache(self):
-        destination = os.path.join(self.cache_path, self.hostdir)
+        destination = os.path.join(self.cache_path, self.target_host)
```

Comment:
@asteven Why self.target_host instead of self.hostdir at line 242? Wouldn't this try to create directory path that contain target_host?